tests/test_filters.py

```diff
@@ -0,0 +1,294 @@
+        """Test that mean filter maintains NaNs in the output."""
+        arr = np.array([[np.nan, 2, 3], [4, 5, np.nan], [7, 8, 9]], dtype=np.float32)
+        filtered = gu.filters.mean_filter(arr, kernel_size=3)
+        assert np.isnan(filtered[0, 0])
+        assert np.isnan(filtered[1, 2])
+
+    def test_min_max_filter_all_nans(self) -> None:
+        """Test that min/max filters on all-NaN arrays return NaN arrays."""
+        arr = np.full((5, 5), np.nan)
+        filtered_min = gu.filters.min_filter(arr, size=3)
+        filtered_max = gu.filters.max_filter(arr, size=3)
+        assert np.all(np.isnan(filtered_min))
+        assert np.all(np.isnan(filtered_max))
+
+
+class TestDistanceFilter:
+    """Tests for the distance-based outlier filter."""
+
+    landsat_dem = gu.Raster(gu.examples.get_path("everest_landsat_b4")).astype(np.float32)
+
+    def test_dist_filter(self) -> None:
+        """Check that distance filter removes outliers and preserves non-outliers."""
+        ddem = self.landsat_dem.copy()
+
+        count = 1000
+        rng = np.random.default_rng(42)
+        cols = rng.integers(0, high=self.landsat_dem.width - 1, size=count)
+        rows = rng.integers(0, high=self.landsat_dem.height - 1, size=count)
+        ddem.data[rows, cols] = 5000
+
+        filtered_ddem = gu.filters.distance_filter(ddem.data, radius=20, outlier_threshold=50)
+        assert np.all(np.isnan(filtered_ddem[rows, cols]))
+        assert ddem.data.shape == filtered_ddem.shape
+        assert np.all(ddem.data[np.isfinite(filtered_ddem)] == filtered_ddem[np.isfinite(filtered_ddem)])
+
+        ddem.data[rows[:500], cols[:500]] = np.nan
+        filtered_ddem = gu.filters.distance_filter(ddem.data, radius=20, outlier_threshold=50)
+        assert np.all(np.isnan(filtered_ddem[rows, cols]))
+
+    def test_distance_filter_all_nans(self) -> None:
+        """Distance filter should return NaNs if all input is NaNs."""
+        arr = np.full((10, 10), np.nan)
+        filtered = gu.filters.distance_filter(arr, radius=2, outlier_threshold=1)
+        assert np.all(np.isnan(filtered))
+
+    def test_distance_filter_no_outliers(self) -> None:
+        """Ensure no changes occur when no outliers are present."""
+        arr = np.ones((10, 10)) * 10
+        filtered = gu.filters.distance_filter(arr, radius=2, outlier_threshold=5)
+        np.testing.assert_array_equal(arr, filtered)
+
+
+class TestGenericFilter:
+    """Tests for the generic filter API in GeoUtils."""
+
+    landsat_dem = gu.Raster(gu.examples.get_path("everest_landsat_b4")).astype(np.float32)
+
+    def test_generic_filter(self) -> None:
+        """Apply a scipy filter using the generic filter interface."""
+        raster_array = get_array_and_mask(self.landsat_dem)[0]
+        raster_filtered = gu.filters.generic_filter(raster_array, scipy.ndimage.minimum_filter, size=5)
+        assert np.nansum(raster_array) != np.nansum(raster_filtered)
```

Comment:
this is a weak test. Could it be checked against the output of the "min" filter directly?